File: Python/0001-1000/001-100/21-30/28.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        n1 = len(haystack)
        n2 = len(needle)
        if n2 == 0:
            return 0
        for i in range(0, n1-n2+1):
            if haystack[i:i+n2] == needle:
                return i
        return -1

File: Java/0001-1000/001-100/21-30/28.java
class Solution {
    public int strStr(String haystack, String needle) {
        int i = 0, n = haystack.length() - needle.length();
        for (; i <= n; i++) {
            if (haystack.startsWith(needle, i))
                return i;
        }
        return -1;
    }
}